### PR TITLE
runtime/lava: Add method to get job status

### PR DESCRIPTION
**Pull request by nuclearcat**
We need to retrieve job status, to know if it was executed properly or failed.

### CODE REVIEW

File: kernelci/runtime/lava.py [hunk 1]
```diff
@@ -66,10 +66,10 @@ class Callback:
...
     # LAVA job result names
     LAVA_JOB_RESULT_NAMES = {
-        COMPLETE: "PASS",
-        INCOMPLETE: "FAIL",
-        CANCELED: "UNKNOWN",
-        CANCELING: "UNKNOWN",
+        COMPLETE: "pass",
+        INCOMPLETE: "incomplete",
+        CANCELED: "incomplete",
+        CANCELING: "incomplete",
```
Comment (JenySadadia) [inline kernelci/runtime/lava.py:72]: What status does lava use for failed jobs?

Comment (JenySadadia) [reply kernelci/runtime/lava.py:72]: I mean, how would we denote that the job ran successfully but somehow failed?

Comment (nuclearcat) [reply kernelci/runtime/lava.py:72]: Unfortunately LAVA documentation seems not very consitent on this topic. For example here is described LAVA TestJob state machine:
https://lava.readthedocs.io/en/latest/technical-references/state-machine/#testjobs
But there is nothing about results, it has paragraph "Results" in that doc, but it is empty. 
So what i did, i submitted job with invalid rootfs URL, recorded callback details/content, and noted which result code job will return on invalid rootfs url (that will cause job to fail shortly after submission), and what result code returned on "good" job.

Comment (JenySadadia) [reply kernelci/runtime/lava.py:72]: Yes, it's weird that the documentation is empty.
So what result code did you receive?
Maybe @pawiecz is aware of the different lava job results?

**Comment by nuclearcat** [comment]
```
{
    "id": 12724007,
    "status": 3,
    "status_string": "incomplete",
    "state": 5,
    "state_string": "Finished",
    "health": 2,
    "health_string": "Incomplete",
    "submit_time": "2024-02-09 08:01:24.713834+00:00",
    "start_time": "2024-02-09 08:01:40.110214+00:00",
    "end_time": "2024-02-09 08:02:00.006301+00:00",
    "submitter_username": "nuclearcat",
    "failure_comment": null,
    "priority": 10,
    "description": "[kci-staging] 65c5daef4c1455ded8894700 baseline-x86 staging-stable-20240209.0",
    "actual_device_id": "qemu-hetzner-9",
    "definition": --skip-definitio--
    "metadata": [],
    "token": "kernelci-api-token-staging",
    "log": "-- skip-log--
...
```

**Comment by nuclearcat** [comment]
Tested on staging, works well and helping now with broken LAVA templates and etc

**Pull request merged**